### PR TITLE
Use token processor dimensions

### PR DESCRIPTION
**Pull request by tbezman**
Use the width / height from the server to avoid layout shift on the client.

| Before | After |
|--------|--------|
| <video src="https://github.com/gallery-so/gallery/assets/6754223/bb9475b2-8f63-4303-ace8-a50dd433fcf7" /> | <video src="https://github.com/gallery-so/gallery/assets/6754223/5e3f41e6-a452-4003-8bd5-5dec0f04c857" /> |

### CODE REVIEW

File: apps/mobile/src/contexts/GalleryTokenDimensionCacheContext.tsx
Comment (tbezman) [inline apps/mobile/src/contexts/GalleryTokenDimensionCacheContext.tsx:1]: Much better to keep this state local to the component that needs it.

File: apps/mobile/src/components/ProfileView/Tabs/ProfileViewFeaturedTab.tsx [hunk 1]
```diff
@@ -56,7 +54,9 @@ export function ProfileViewFeaturedTab({ queryRef }: ProfileViewFeaturedTabProps
   return (
     <View style={contentContainerStyle}>
       <Tabs.FlashList
-        estimatedItemSize={300}
+        getItemType={(item) => item.kind}
+        keyExtractor={(item) => item.key}
+        estimatedItemSize={100}
```
Comment (tbezman) [inline apps/mobile/src/components/ProfileView/Tabs/ProfileViewFeaturedTab.tsx:59]: should have the `getItemType` and `keyExtractor` on all of these. `estimatedItemSize` helps with funky mounting behavior

File: apps/mobile/src/components/ProfileView/Tabs/ProfileViewFeaturedTab.tsx [hunk 2]
```diff
@@ -39,13 +35,15 @@ export function ProfileViewFeaturedTab({ queryRef }: ProfileViewFeaturedTabProps
...
+      return { items: [], stickyIndices: [] };
+    }
 
-  const { items, stickyIndices } = createVirtualizedGalleryRows({
-    galleryRef: user.featuredGallery,
-  });
+    return createVirtualizedGalleryRows({
+      galleryRef: user.featuredGallery,
+    });
+  }, [user?.featuredGallery]);
```
Comment (tbezman) [inline apps/mobile/src/components/ProfileView/Tabs/ProfileViewFeaturedTab.tsx:46]: had anxiety about this not being memoized.

File: apps/mobile/src/components/ProfileView/ProfileView.tsx [hunk 3]
```diff
@@ -77,42 +76,40 @@ export function ProfileView({ queryRef, shouldShowBackButton }: ProfileViewProps
       </View>
 
       <View className="flex-grow">
-        <GalleryTokenDimensionCacheProvider>
```
Comment (tbezman) [inline apps/mobile/src/components/ProfileView/ProfileView.tsx:80]: just removed the provider

File: apps/mobile/src/components/Gallery/GalleryTokenPreview.tsx [hunk 4]
```diff
@@ -40,36 +48,50 @@ export function GalleryTokenPreview({ tokenRef, containerWidth }: GalleryTokenPr
...
       });
+    }
 
-      return result;
+    if (measuredDimensions) {
+      return fitDimensionsToContainerContain({
+        container: { width: containerWidth, height: screenDimensions.height / 2 },
+        source: measuredDimensions,
+      });
     }
```
Comment (tbezman) [inline apps/mobile/src/components/Gallery/GalleryTokenPreview.tsx:85]: In this part, we're just trying to use the server dimensions as a primary source of truth, and then fallback to the measured dimensions.

**Review by kaitoo1 — APPROVED** [review]

**Pull request merged**